### DISCUSSION TITLE
Public rows/search api appears to be getting from development database instead of production database

### DISCUSSION THREAD

**Discussion by magickaito** [Budibase/budibase#6153] [Help] [answered]
I am trying out one self-hosted budibase application on DigitalOcean, using internal tables.

I noticed the following:
- Let's say there is a Table with 3 records in production
- User removes 1 record from the table in production, now 2 records left.
- Public API http://IP:10000/api/public/v1/tables/TABLEID/rows/search **still getting 3 records** in the reponse
- Now navigate to budibase builder web interface, dev database sync, now showing 2 records on builder web interface
- Now call public API again, response will be 2 records.

So public API response remains incorrect until someone visits the budibase builder interface? 
Why is the public API response appears to be returning results from dev database instead of production? 

For the app-id, I followed the documentation to get it from the builder path: https://docs.budibase.com/docs/public-api, so the app ID is something like app_dev_.... Is this the reason? But Is there a way to tell the public API that we want results from production?

This is the Node.js javascript code that tries to get records from public API:

```
function getInspectionsFromBudibasePublicAPI(rid, callback){
	var options = {
	  'method': 'POST',
	  'gzip': true,
	  'url': 'http://IP:10000/api/public/v1/tables/ta_10bc5cxxxxxxxx/rows/search',
	  'headers': {
	    'x-budibase-app-id': 'app_dev_fe0axxxxxxx',
	    'Content-Type': 'application/json',	 
	    'Accept': 'application/json',
	    'x-budibase-api-key': 'xxxxxx'	    
	  },
	  body: JSON.stringify({"query":{"string":{"rid":rid}}})

	};
	request(options, function (error, response) {
	  if (error) throw new Error(error);
	  console.log("getInspectionsFromBudibasePublicAPI", response.body);
	  callback(JSON.parse(response.body));
	});
}
```

**Comment by melohagan** [comment, marked as answer]
Hi @magickaito 

If you want to retrieve results from production, then you'll need to use the app id for your published app.

It should be identical, just without the `_dev_` part. 

E.g. 
- builder: `app_dev_bb_afdf15d335df499da0932ed736edf7e3`
- published: `app_bb_afdf15d335df499da0932ed736edf7e3`

**Reply by magickaito** [reply]
learnt another new thing today. thanks!

**Reply by melohagan** [reply]
I've added a note into the docs as well.
